modules/wallet/transactionbuilder.go

```diff
@@ -382,6 +382,26 @@ func (tb *transactionBuilder) FundSiafunds(amount types.Currency) error {
 	return nil
 }
 
+// UnconfirmedParents returns any unconfirmed parents the transaction set that
+// is being built by the transaction builder could have.
+func (tb *transactionBuilder) UnconfirmedParents() (parents []types.Transaction) {
+	addedParents := make(map[types.TransactionID]struct{})
+	for _, p := range tb.parents {
+		for _, sci := range p.SiacoinInputs {
+			tSet := tb.wallet.tpool.TransactionSet(crypto.Hash(sci.ParentID))
+			for _, txn := range tSet {
+				txnID := txn.ID()
+				if _, exists := addedParents[txnID]; exists {
+					continue
+				}
+				addedParents[txnID] = struct{}{}
```

Comment:
technically this is over-generous I think? You are adding every txn in the set, which may include children txns as well.

I think that it's probably okay for what we're doing, especially if the txn isn't completely published yet, but if the txn is completely published and you are still using the builder then you could get extra txns.

not sure if we want to handle that